pytensor/tensor/rewriting/linalg.py

```diff
@@ -539,3 +540,64 @@ def svd_uv_merge(fgraph, node):
                     or len(fgraph.clients[cl.outputs[2]]) > 0
                 ):
                     return [cl.outputs[1]]
+
+
+def _find_solve_with_eye(node):
```

Comment:
Docstring and return typehint here as well (no need to typehint `node`; that's a bit of a rabbit hole, but add `-> bool`)